chainer/training/trainer.py

```diff
@@ -0,0 +1,285 @@
+      whether the extension should be executed at the iteration by returning a
+      boolean value. In most cases, :class:`IntervalTrigger` is used, in which
+      case users can simply specify a tuple of the interval length and its
+      unit, like ``(1000, 'iteration')`` or ``(1, 'epoch')``.
+    - The order of execution of extensions is determined by their priorities.
+      Extensions of higher priorities are invoked earlier. There are three
+      standard values for the priorities:
+
+      - ``PRIORITY_WRITER``. This is the priority for extensions that write
+        some records to the :attr:`observation` dictionary. It includes cases
+        that the extension directly adds values to the observation dictionary,
+        or the extension uses the :func:`chainer.report` function to report
+        values to the observation dictionary.
+      - ``PRIORITY_EDITOR``. This is the priority for extensions that edit the
+        :attr:`observation` dictionary based on already reported values.
+      - ``PRIORITY_READER``. This is the priority for extensions that only read
+        records from the :attr:`observation` dictionary. This is also suitable
+        for extensions that do not use the :attr:`observation` dictionary at
+        all.
+
+    - Extensions with ``invoke_before_training`` flag on are also invoked at
+      the beginning of the training loop. Extensions that update the training
+      status (e.g., changing learning rates) should have this flag to be True
+      to ensure that resume of the training loop correctly recovers the
+      training status.
+
+    The current state of the trainer object and objects handled by the trainer
+    can be serialized through the standard serialization protocol of Chainer.
+    It enables us to easily suspend and resume the training loop.
+
+    .. note::
+       The serialization does not recover everything of the training loop. It
+       only recovers the states which change over the training (e.g.
+       parameters, optimizer states, the batch iterator state, extension
+       states, etc.). You must initialize the objects correctly before
+       deserializing the states.
+
+       On the other hand, it means that users can change the settings on
+       deserialization. For example, the exit condition can be changed on the
+       deserialization, so users can train the model for some iterations,
+       suspend it, and then resume it with larger number of total iterations.
+
+    During the training, it also creates a :class:`~chainer.Reporter` object to
+    store observed values on each update. For each iteration, it creates a
+    fresh observation dictionary and stores it in the :attr:`observation`
+    attribute.
+
+    Links of the target model of each optimizer are registered to the reporter
+    object as observers, where the name of each observer is constructed as the
+    format ``<optimizer name><link name>``. The link name is given by the
```

Comment:
Yes, because the link's name always begins with slash ``/``, and it acts like a delimiter.